### PR TITLE
fix scaling pd with Advanced StatefulSet

### PR DESCRIPTION
**Pull request by cofyc**
Signed-off-by: Yecheng Fu <fuyecheng@pingcap.com>

### What problem does this PR solve? 

fix scaling pd with Advanced StatefulSet

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e/tidbcluster/serial.go [hunk 1]
```diff
@@ -216,6 +216,12 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 					replicas:    3,
 					deleteSlots: sets.NewInt32(0, 3),
 				},
+				{
+					name:        "Scaling out pd from 3 to 5 by adding pods 5 and 6",
+					component:   "pd",
+					replicas:    5,
+					deleteSlots: sets.NewInt32(0, 3),
+				},
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go]: new case "scaling out while pods are not consecutive"

File: tests/e2e/tidbcluster/serial.go [hunk 2]
```diff
@@ -193,16 +193,16 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
...
-					deleteSlots: sets.NewInt32(),
+					deleteSlots: sets.NewInt32(1),
 				},
 				{
-					name:        "Scaling out tidb from 2 to 4 by adding pods 3 and 4",
+					name:        "Scaling tidb from 2 to 4 by deleting pods 2 and adding pods 3, 4 and 5",
 					component:   "tidb",
 					replicas:    4,
-					deleteSlots: sets.NewInt32(1),
+					deleteSlots: sets.NewInt32(1, 2),
```
Comment (cofyc) [inline tests/e2e/tidbcluster/serial.go]: cover some corner cases
